logpuzzle.py

```diff
@@ -40,14 +76,26 @@ def download_images(img_urls, dest_dir):
     with an img tag to show each local image file.
     Creates the directory if necessary.
     """
-    # +++your code here+++
-    pass
+    os.mkdir(dest_dir)
+
+    with open('index.html', 'w')as index:
+        img_list = []
+        for i, url in enumerate(img_urls):
+            urllib.urlretrieve(url, 'img{}'.format(i))
+            shutil.move('img{}'.format(i), '{}'.format(dest_dir))
+            img_list.append(create_html_img("'img{}'".format(i)))
+        shutil.move("index.html", dest_dir)
+        index.write(html_temp.format('\n'.join(img_list)))
```

Comment:
Joining on newline is causing a margin around image elements that's preventing the image slices from rendering as an uninterupted image.